core/src/main/java/org/geysermc/geyser/command/CommandRegistry.java

```diff
@@ -275,8 +278,12 @@ private void buildRootCommand(String permission, HelpCommand help) {
         cloud.command(builder.handler(context -> {
             GeyserCommandSource source = context.sender();
             if (!source.hasPermission(help.permission())) {
-                // delegate if possible - otherwise we have nothing else to offer the user.
-                source.sendLocaleString(ExceptionHandlers.PERMISSION_FAIL_LANG_KEY);
+                if (STANDALONE_COMMAND_MANAGER && source instanceof GeyserSession session) {
+                    session.sendCommand(context.rawInput().input());
+                } else {
+                    // delegate if possible - otherwise we have nothing else to offer the user.
+                    source.sendLocaleString(ExceptionHandlers.PERMISSION_FAIL_LANG_KEY);
+                }
```

Comment:
consider this instead (I can't provide it as suggestion because you haven't changed all of it haha)
```java
        cloud.command(builder.handler(context -> {
            GeyserCommandSource source = context.sender();
            if (source.hasPermission(help.permission())) {
                // Delegate to help if possible
                help.execute(source);
            } else if (STANDALONE_COMMAND_MANAGER && source instanceof GeyserSession session) {
                // If we are on an appropriate platform, forward the command to the backend
                session.sendCommand(context.rawInput().input());
            } else {
                source.sendLocaleString(ExceptionHandlers.PERMISSION_FAIL_LANG_KEY);
            }
        }));
```